### DISCUSSION TITLE
Allow `@supports` to work custom `screens`

### DISCUSSION THREAD

**Discussion by nghiepdev** [tailwindlabs/tailwindcss#9562] [Ideas]
Currently, we can use `raw` to custom advance media queries, But still no support `@supports` key:
```js
module.exports = {
  theme: {
    extend: {
      screens: {
        touch: {raw: '(hover: none) and (pointer: coarse)'},
        standalone: {raw: '(display-mode: standalone)'},
      },
    },
  },
}
```

Request features look like this:
```js
module.exports = {
  theme: {
    extend: {
      screens: {
        ios: {
          supports: '(-webkit-touch-callout: none)',
        },
      },
    },
  },
}
```
----------
**UPDATED: 4 ways to resolve**
```html
<div class="[@supports(-webkit-touch-callout:none)]:text-red-500">1</div>
```
```html
<div class="supports-[-webkit-touch-callout:none]:text-red-500">2</div>
```
```js
plugins: [
  plugin(function({ addVariant }) {
    addVariant("ios", "@supports(-webkit-touch-callout:none)")
  })
]
<div class="ios:text-red-500" />
```
```js
theme: {
  supports: {
    ios: '-webkit-touch-callout: none'
  }
}
```

**Comment by wongjn** [comment]
You could look at trying out the insiders build (#9504) which has a [new variant for `@supports` rules](https://github.com/tailwindlabs/tailwindcss/discussions/9504#new-variant-for-supports-rules). Otherwise until v3.2, you could use an [arbitrary variant](https://tailwindcss.com/docs/hover-focus-and-other-states#using-arbitrary-variants) like `[@supports(-webkit-touch-callout:none)]:text-black`.

**Reply by nghiepdev** [reply]
Wow. Thanks @wongjn
